incubator/hnc/hack/prow-e2e/Dockerfile

```diff
@@ -1,4 +1,11 @@
-FROM golang:1.14.2-alpine
+# krte includes everything needed to run Docker-in-Docker, as well as cleaning
+# it up cleanly. Note that Prow uses containerd so typically, no Docker daemon
+# will be running on the node without using the contents of this image!
+#
+# This includes Go 1.15.7 and kubectl 1.17
+#
+# Source: https://github.com/kubernetes/test-infra/tree/master/images/krte
+FROM gcr.io/k8s-testimages/krte:v20210129-3799a64-master
```

Comment:
Do we need any docs around tracking the proper image to use? As in, how/when would I know we should change this?

Alternatively, it looks like there are a few tracking tags existing (although using tracking tags can cause other headaches):

- latest-master
- latest-1.17
- latest-1.18
- ...

Reply:
I decided to stick to the existing one but added a bunch of docs in the Dockerfile and README about when to upgrade to a later version. I doubt that we're very sensitive to the exact version but I do think it makes sense to pin it just in case.